src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -348,13 +370,12 @@ public CountingReadFilter getMergedCountingReadFilter(final SAMFileHeader samHea
         Utils.nonNull(samHeader);
         Utils.nonNull(aggregateFunction);
 
-        if (disableAllReadFilters) {
-            return aggregateFunction.apply(null, samHeader);
-        }
-
         // start with the tool's default filters in the order they were specified, and remove any that were disabled
         // on the command line
-        final List<ReadFilter> finalFilters = toolDefaultReadFilterNamesInOrder
+        // if --disableToolDefaultReadFilters is specified, just initialize an empty list with initial capacity of user filters
+        final List<ReadFilter> finalFilters = (disableToolDefaultReadFilters)
+                ? new ArrayList<>(userReadFilterNames.size())
+                : toolDefaultReadFilterNamesInOrder
                 .stream()
                 .filter(s -> !isDisabledFilter(s))
                 .map(s -> toolDefaultReadFilters.get(s))
```

Comment:
I think this was much clearer before (I realize it had to change to account for the new meaning of disableToolDefaultReadFilters). I guess its partly a style thing, but the extra parens and lack of indentation makes it hard to read; I think its much easier to parse like this:

```
        final List<ReadFilter> finalFilters = 
                disableToolDefaultReadFilters ?
                        new ArrayList<>(userReadFilterNames.size()) :
                        toolDefaultReadFilterNamesInOrder
                            .stream()
                            .filter(s -> !isDisabledFilter(s))
                            .map(s -> toolDefaultReadFilters.get(s))
                            .collect(Collectors.toList());
```